app/models/user.rb

```diff
@@ -14,10 +14,13 @@ class User < ApplicationRecord
           :validatable,
           :lockable
 
+  before_validation :generate_uuid
```

Comment:
have generally handled uuid generation on the database directly rather than the model

Reply:
currently this is not a UUID. i'd name it `generate_unique_ref` or somesuch